### PR TITLE
1828 - M&H presidency exchanges

### PR DESCRIPTION
**Pull request by crericha**
- Exchange for one share of the presidency (cash for the other after parring)
- Add logic to trigger parring of the corporation after the exchange
- If exchange causes corporation to float, it operates in the next operating in which no corporations have yet operated (could be the current OR)

### CODE REVIEW

File: assets/app/view/game/buy_sell_shares.rb [hunk 1]
```diff
@@ -138,7 +140,15 @@ def render_exchanges
...
             if ability.from.include?(:ipo)
-              children.concat(render_share_exchange(@ipo_shares, company, source: @game.ipo_name(@corporation)))
+              children.concat(render_share_exchange(@ipo_shares.reject(&:president),
+                                                    company,
+                                                    source: @game.ipo_name(@corporation)))
+              if @game.exchange_for_partial_presidency? && (presidency_share = @ipo_shares.find(&:president))
+                children.concat(render_share_exchange([presidency_share],
+                                                      company,
+                                                      source: 'Presidency',
+                                                      partial_percent: 100 / presidency_share.num_shares))
```
Comment (tobymao) [inline assets/app/view/game/buy_sell_shares.rb:150]: why is this logic in the view? shouldn't it be a part of the game?

Comment (crericha) [reply assets/app/view/game/buy_sell_shares.rb:150]: Do you mean the partial_percent bit or the whole thing?

Comment (tobymao) [reply assets/app/view/game/buy_sell_shares.rb:150]: partial percent bit

Comment (crericha) [reply assets/app/view/game/buy_sell_shares.rb]: Move to game class.

**Review by tobymao — APPROVED** [review]

**Pull request merged**